requerimiento 1 back

defryd opened the pull request:
edición de la clase ProyectoController del paquete impl con el avance del requerimiento 1

defryd committed:
requerimiento 1 back

1151704 commented:
1. Solo debe existir una clase main en el sistema, para realizar pruebas debe hacerlo en el paquetes de pruebas.
2. Revisar la sangría del código, para que tenga buena legibilidad.
3. No está implementando el patrón MVC, el constructor no puede ser modificado, al menos que sea necesario, pero en su caso lo modificó únicamente para hacer funcionar el main.
4. Los métodos debe estar documentados.
5. La implementación que está tratando de dar, ya está realizada en el controlador del menú.

defryd committed:
Merge branch 'main' into main

1151704 commented:
No he recibido más comentarios acerca del commit, el día de mañana procederé a cerrar el request.

defryd committed:
Merge branch 'Arquitectura-de-Software-UFPS-2022-I:main' into main

defryd committed:
commit arreglo Req 3 y 4

requerimiento 3 listo
requerimiento 4 en proceso un 40%

defryd committed:
Merge branch 'main' of https://github.com/Arquitectura-de-Software-UFPS-2022-I/python-poo-gui

defryd committed:
Merge branch 'main' of https://github.com/Izequ/python-poo-gui

defryd committed:
edit ventana ayuda

se edito solo una linea de la ventana ayuda

defryd committed:
Merge branch 'main' into main

defryd committed:
Merge branch 'Arquitectura-de-Software-UFPS-2022-I:main' into main

defryd committed:
Merge branch 'Arquitectura-de-Software-UFPS-2022-I:main' into main

defryd committed:
commit de req 3 y 4

Culminación del requerimiento 3 y 4.
También se arreglo un error que tenia en mi repositorio con el requerimiento 2.

defryd committed:
Merge branch 'main' of https://github.com/Izequ/python-poo-gui

1151704 commented on src/main/java/ufps/arqui/python/poo/gui/controllers/scan.py:
```diff
@@ -0,0 +1,169 @@
+import os, json
```
Este archivo no debería estar aquí

defryd replied:
eliminado

1151704 commented on src/main/java/ufps/arqui/python/poo/gui/controllers/impl/ProyectoController.java:
```diff
@@ -1,5 +1,18 @@
 package ufps.arqui.python.poo.gui.controllers.impl;
 
+
```
Imports No utilizados, favor de eliminarlos.

defryd replied:
arreglado

1151704 commented on src/main/java/ufps/arqui/python/poo/gui/views/IPanelMenu.java:
```diff
@@ -20,4 +20,6 @@ public interface IPanelMenu extends Observer, IPanelView {
      */
     void modalCrearProyecto(String nombre, String directorio)  throws IOException;
+
+    void modalAbrirProyecto(String nombre, String directorio) throws  IOException;
```
Faltante Comentario en método

defryd replied:
comentario agregado

1151704 commented:
![image](https://user-images.githubusercontent.com/44126200/175350268-e4c51a1f-894f-4272-b2b9-bf0b40a709c2.png)
El diseño está con errores, adicionalmente, cuando se abre un proyecto anterior, este debe traer el nombre del proyecto, no se debe pedir el nombre.
Al crear un nuevo proyecto, se debe guardar persistentemente el nombre del proyecto, para que al momento de abrir un proyecto este pueda ser consultado.

defryd committed:
Merge branch 'Arquitectura-de-Software-UFPS-2022-I:main' into main

defryd commented:
requerimiento 4, creación de la ModalAyuda y el evento del abrir manual de usuario

defryd reviewed (commented):
requerimiento 4, creación de la ModalAyuda y el evento del abrir manual de usuario

1151704 commented:
Modal de ayuda mal estructurada, ya que presenta grandes espacios en blanco, no presenta sangria y datos imcompletos, adicionalmente el botón del manual está es español/ingles.
![image](https://user-images.githubusercontent.com/44126200/175941014-66b7b1f5-5aef-488b-a45e-75bdd8c9434e.png)

1151704 commented on src/main/java/ufps/arqui/python/poo/gui/views/impl/ModalAyuda.java:
```diff
@@ -0,0 +1,240 @@
+    }
+    public void mostrarPDF(){
+        try {
+                 File path = new File ("./src/main/java/ufps/arqui/python/poo/gui/views/impl/Manual_test.pdf");
```
No es la mejor forma para acceder al pdf, todos los recursos deben ir dentro de: src/main/resources

defryd replied:
método modificado

1151704 commented on src/main/java/ufps/arqui/python/poo/gui/views/impl/ModalAbrirProyecto.java:
```diff
@@ -0,0 +1,234 @@
+import java.io.IOException;
+
+/**
+ *
```
Sn documentación de la clase

defryd replied:
documentado

1151704 commented on src/main/java/ufps/arqui/python/poo/gui/views/impl/ModalAbrirProyecto.java:
```diff
@@ -0,0 +1,234 @@
+    }
+
+    private void addEvents() {
+        DocumentListener docEvent = new DocumentListener() {
```
Invocación del evento, pero sin ningún uso

defryd replied:
solucionado

1151704 commented on src/main/java/ufps/arqui/python/poo/gui/views/impl/ModalAyuda.java:
```diff
@@ -0,0 +1,240 @@
+import ufps.arqui.python.poo.gui.views.IPanelMenu;
+
+/**
+ *
```
Sin documentación de la clase

defryd replied:
solucionado

1151704 commented on src/main/java/ufps/arqui/python/poo/gui/views/impl/ModalAyuda.java:
```diff
@@ -0,0 +1,240 @@
+    }
+
+    private void addEvents() {
+        DocumentListener docEvent = new DocumentListener() {
```
Invocación del evento, pero sin utilización

defryd replied:
borrado

defryd committed:
Arreglo de requerimientos 3 y 4

arreglo visual del requerimiento 3 y 4, edición del metodo de abrir archivo.

defryd committed:
Delete scan.py

defryd committed:
documentación

documentación de las clases ModalAyuda y ModalAbrirProyecto

1151704 commented:
Buen trabajo, subiré sus cambios a la rama principal.

Quedó pendiente la persistencia del nombre del proyecto, el cual al momento de crear uno se debe almacenar y al momento de abrir un proyecto existente mostrarlo.
